StudioCMS Beta.16 release docs (#73)
withstudiocms/docs: src/content/docs/en/guides/contributing/translations.mdx

---
i18nReady: true
title: Translations
description: Learn how to contribute translations to StudioCMS
sidebar:
  order: 2
  badge:
    text: Updated
    variant: success
---

import ReadMore from '~/components/ReadMore.astro';
import { FileTree } from '@astrojs/starlight/components';

StudioCMS is a global project, and we want to make it accessible to everyone. If you are fluent in multiple languages, you can help us translate StudioCMS into your language.

## Dashboard i18n

Current translation status:

<img src="https://badges.awesome-crowdin.com/translation-16993424-776180-update.png" alt="Translation status" />

> Visit [our i18n dashboard](https://i18n.studiocms.dev) to help translate StudioCMS into your language. If your language is not listed, you can add it within the dashboard.

Reviewer: ```suggestion
Visit [our i18n dashboard](https://i18n.studiocms.dev) to help translate the StudioCMS package into your language. If your language is not listed, you can add it within the dashboard.
```

Author: "packages"?

Reviewer: Okay, good to know that all your packages are translated! Thanks! 👍

> If you prefer to contribute translations directly to the repository, the translations are stored in the [`packages/studiocms/src/lib/i18n/translations`](https://github.com/withstudiocms/studiocms/tree/main/packages/studiocms/src/lib/i18n/translations/) directory. You can find the English translations in the [`en-us.json`](https://github.com/withstudiocms/studiocms/blob/main/packages/studiocms/src/lib/i18n/translations/en-us.json) file.

Reviewer: ```suggestion
If you prefer to contribute translations directly to the repository, the translations are stored in the [`packages/studiocms/src/lib/i18n/translations`](https://github.com/withstudiocms/studiocms/tree/main/packages/studiocms/src/lib/i18n/translations/) directory. You can find the English translations in the [`en-us.json`](https://github.com/withstudiocms/studiocms/blob/main/packages/studiocms/src/lib/i18n/translations/en-us.json) file, which will override the Crowdin translation.
```

Reviewer: I'm not sure how it works exactly, but as a reader it is kinda confusing having two sources of truth...

Author: it does not "override" it syncs :sweat_smile:

Reviewer: Perfect, thanks for explaining! I updated the suggestion regardingly, let me know what you think!

<ReadMore>

> StudioCMS uses [Crowdin](https://crowdin.com/) for managing translations on top of GitHub. If you are new to Crowdin, you can find the [For Translators guide](https://support.crowdin.com/for-translators/) on their website.

Reviewer: ```suggestion
The StudioCMS core package uses [Crowdin](https://crowdin.com/) for managing translations on top of GitHub. If you are new to Crowdin, you can find the [For Translators guide](https://support.crowdin.com/for-translators/) on their website.
```

Author: This will actually not be true for other plugins and packages within the monorepo... maybe we can remove the core part? (i.e. "StudioCMS packages use Crowdin...")

Reviewer: You can decide, wheter to keep the `All ` or remove it...

</ReadMore>

Once the translations have been added, they will be added to the [StudioCMS i18n configuration](https://github.com/withstudiocms/studiocms/blob/main/packages/studiocms/src/lib/i18n/index.ts#L8) and will be available in the next release.

> ## Translations

Reviewer: ```suggestion
## Docs Translations
```

StudioCMS is actively working on translating their docs into multiple languages.

> Currently, we are aiming to translate the Astro documentation into the following languages:

Author: ```suggestion
Currently, we are aiming to translate the StudioCMS documentation into the following languages:
```

### Structure of the documentation

The docs repository contains three different kinds of translatable content: docs pages, and UI strings. The following file tree shows where to find these in the repository.

<FileTree>

- ...
- src/
  - ...
  - content/
    - docs/ MDX page content, one directory per language
      - de/
      - en/
      - es/
      - ...
    - i18n/ UI string translations, one file per language
      - de.json
      - en.json
      - es.json
      - ...
  - starlight-sidebar/ UI string translations for Sidebar, one file per language
    - de.json
    - en.json
    - es.json
    - ...
- astro.config.ts
- package.json
- ...

</FileTree>

### Translation Process

Ready to dive in? Here's how to find what you need in the repo:

1. **Page-specific content** (page title, body copy, etc.)

   ➤ Go to `src/content/docs/{language}/{page-slug}.mdx`

2. **Shared text** (search modal, right sidebar, “next article” links, tutorials, etc.)

   ➤ Go to `src/content/i18n/{language}.json`

3. **Primary Sidebar Translations**

   ➤ Go to `src/starlight-sidebar/{language}.json`

> Once you've located your target file, peek at the [Translation Tracker](https://i18n.docs.astro.build/) and existing PRs, maybe someone might already be on it. If it's free, shout out in the `docs-i18n` channel in Discord so nobody duplicates work!

Reviewer 2: Wrong link:
```suggestion
Once you've located your target file, peek at the [Translation Tracker](https://i18n.docs.studiocms.dev/) and existing PRs, maybe someone might already be on it. If it's free, shout out in the `docs-i18n` channel in Discord so nobody duplicates work!
```
